README.md

```diff
@@ -130,83 +134,98 @@ const ipld = new Ipld({
 })
 ```
 
-##### `options.loadFormat(codec, callback)`
+##### `options.loadFormat(codec)`
 
 | Type | Default |
 |------|---------|
-| `Function` | `null` |
+| `async Function` | `null` |
 
-Function to dynamically load an [IPLD Format](https://github.com/ipld/interface-ipld-format). It is passed a string `codec`, the multicodec of the IPLD format to load and a callback function to call when the format has been loaded. e.g.
+Function to dynamically load an [IPLD Format](https://github.com/ipld/interface-ipld-format). It is passed a `codec`, the multicodec code of the IPLD format to load and returns an IPLD Format implementation. For example:
 
 ```js
+const multicodec = require('multicodec')
 const ipld = new Ipld({
-  loadFormat (codec, callback) {
-    if (codec === 'git-raw') {
-      callback(null, require('ipld-git'))
+  async loadFormat (codec) {
+    if (codec === multicodec.GIT_RAW) {
+      return require('ipld-git')
     } else {
-      callback(new Error('unable to load format ' + codec))
+      throw new Error('unable to load format ' + multicodec.print[codec])
     }
   }
 })
 ```
 
-### `.put(node, options, callback)`
+### `.put(nodes, options)`
+
+> Stores the given IPLD Nodes of a recognized IPLD Format.
+
+ - `nodes` (`Iterable<Object>`): deserialized IPLD nodes that should be inserted.
```

Comment:
...or just allow passing a `Block`

Reply:
Ya, I think that we should allow Block instances in this API so that users can handle whatever mixing and matching of encodings they want. In fact, I’m tempted to *only* have that be the case and just get this API out of this business of block encoding but I’ll leave that decision to @vmx :)

Reply:
I don't see how passing in a `Block` would help. A `Block` is just data + a `CID`. And we don't know the `CID` yet (this feature was explicitly removed).

Reply:
@vmx this API takes node data, in the form of an iterable, encodes it and stores it. In the case where I have *already* encoded the data I’d like to pass that in to *just* be stored. This also solves Alan’s case, which is that he has an iterable of data in different formats (raw, cbor) and needs to store them all.

Another option would be to add another API called `store()` that skips the encoding step and just stores the blocks. This would also likely be used internally by `put()` after it encodes each block.

Reply:
There was some discussions around it, that removing passing in a `cid` as parameter makes sense as it should always be calculated so you make sure you don't store data with a CID that doesn't match.

I think storing something without parsing and calculating a CID would be more like the Block API and I would keep it separate from IPLD.

Reply:
> An idea: what if each input item was `{ value, format?, hashAlg?, version? }`? You'd still pass options but could configure the node being inserted on a per-node level.

That sounds OK to me. It can be added later on (without breaking the existing API) when it turns out that we need it.